### ISSUE TITLE
Final zoom value breaks scale bar value

### ISSUE THREAD

**Issue by dalthviz** [napari/napari#6055]
## To Reproduce

1. Open an example image (for example `Cell`)
2. Activate scale bar (`View > Scale Bar > Scale Bar Visible`)
3. Set units for the scale bar (from the napari-console run: `viewer.scale_bar.unit = "um"` )
4. Zoom In as far as you can

### Preview

![zoom_in](https://github.com/napari/napari/assets/16781833/7024807d-cb1a-4475-a471-0cc52f697209)

With the last possible zoom in action the scale bars goes from `2fm` to `200nm`

## Expected behavior

The scale bar value should show the lowest possible value when doing the last possible zoom in action instead of jumping to a higher value

## Environment

<details>

```
napari: 0.5.0a2.dev231+g7689d09a
Platform: Windows-10-10.0.19045-SP0
Python: 3.10.9 | packaged by conda-forge | (main, Jan 11 2023, 15:15:40) [MSC v.1916 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.9
NumPy: 1.24.3
SciPy: 1.9.3
Dask: 2022.10.2
VisPy: 0.12.1
magicgui: 0.6.0
superqt: unknown
in-n-out: 0.1.5
app-model: 0.1.4
npe2: 0.7.0

OpenGL:
- GL version: 4.5.0 - Build 25.20.100.6444
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1366x768, scale 1.0
- screen 2: resolution 1600x900, scale 1.0

Settings path:
- C:\Users\dalth\AppData\Local\napari\napari-dev_4454fdbf170edefe23827a4bf512389a6806af69\settings.yaml
```

</details>

## Additional context

This was discovered while working on PR #6048 See https://github.com/napari/napari/pull/6048#issuecomment-1631364139

**Comment by brisvag** [comment]
I have a feeling that it's because the camera zoom doesn't have a *smart* stopping mechanism, and instead it stops because of some numerical overflow, which is unfortunately sent to the scalebar as well.

**Comment by brisvag** [comment]
yep, the camera zoom jumps like crazy. This is a vispy issue I think, we just rely on its callbacks for camera controls.

**Comment by brisvag** [comment]
Actually, it's a combination :P
